src/app/utils/BountyUtils.ts

```diff
@@ -262,6 +269,201 @@ const BountyUtils = {
+                    const draftCardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, draftChannel);
+                    await draftCardMessage.delete();
+                    bounty.canonicalCard = undefined;
+                }
+            }
+            if (!!bounty.canonicalCard) {  // If we still have an existing card, just edit it, remove old reactions
+                bountyChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
+                cardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, bountyChannel);
+                await cardMessage.edit(cardEmbeds);
+                await cardMessage.reactions.removeAll();
+            } else {  // Otherwise create it. Put it in the passed in channel, or customer channel by default
+                if (!bountyChannel) bountyChannel = await DiscordUtils.getBountyChannelfromCustomerId(bounty.customerId);
+                cardMessage = await bountyChannel.send(cardEmbeds);
+            }
+        }
+        reacts.forEach(react => {
+            cardMessage.react(react);
+        });
+
+
+        // Delete old cards if they exist. Notify user of new card location with link
+
+        if (bounty.discordMessageId) {
+            await this.notifyAndRemove(bounty.discordMessageId, await DiscordUtils.getBountyChannelfromCustomerId(bounty.customerId), cardMessage.url);
+        }
+        if (!!bounty.creatorMessage) {
+            await this.notifyAndRemove(bounty.creatorMessage.messageId, await DiscordUtils.getTextChannelfromChannelId(bounty.creatorMessage.channelId), cardMessage.url);
+        }
+        if (!!bounty.claimantMessage) {
+            await this.notifyAndRemove(bounty.claimantMessage.messageId, await DiscordUtils.getTextChannelfromChannelId(bounty.claimantMessage.channelId), cardMessage.url);
+        }
+         
+        // Store the card location in the bounty, remove the old cards
+        const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne({ _id: new mongo.ObjectId(bounty._id) }, {
+            $unset: {
+                claimantMessage: "",
+                creatorMessage: "",
+                discordMessageId: "",
+            },
+            $set: { canonicalCard: {
+                        messageId: cardMessage.id,
+                        channelId: cardMessage.channelId,
+                    },
+            },
+        });
+    
+
+        return cardMessage;
+    
+    },
+
+    async notifyAndRemove(messageId: string, channel: TextChannel, cardUrl: string): Promise<any> {
+        let message: Message;
+        try {
+            message = await DiscordUtils.getMessagefromMessageId(messageId, channel)
+        } catch {
+            Log.error(`Old bounty card message <${messageId}> not found in channel <${channel.id}>`);
+        }
+        if (!!message) message.delete();
+        channel.send(`Bounty card has been moved: ${cardUrl}`);
```

Comment:
I think you'll want to use await: `await message.delete();` and `await channel.send('')`

Reply:
Good catch